### PR TITLE
esp32 port: expose adjtime(3) as utime.adjtime([(sec,usec)])

### PR DESCRIPTION
**Pull request by wieck**
utime.adjtime() can be used to slew the system clock by a
given adjustment value (as an optional (sec, usec) tuple).
The system clock is not abruptly changed, but gradually
adjusts the boot time. The call immediately returns, but
a correction by 1 second will actually take 64 seconds in
the background. The system clock remains monotonic during
this time but will appear faster/slower. Called without
an argument or None it will return the remaining adjustment
from a previous call. If called with an adjustment value a
previously running adjustment is cancelled without being
undone.

### CODE REVIEW

**Comment by wieck** [comment]
Discussion at https://forum.micropython.org/viewtopic.php?f=18&t=7403

**Comment by tve** [comment]
IMHO this needs a doc update commit. Would love to see it merged then! :+1:

File: docs/library/utime.rst [hunk 1]
```diff
@@ -227,3 +227,13 @@ Functions
       hardware also lacks battery-powered RTC, so returns number of seconds
       since last power-up or from other relative, hardware-specific point
       (e.g. reset).
+
+
+.. function:: adjtime([(secs, usecs)])
+
+   This function is only available in the esp32 port. It uses the API's
```
Comment (tve) [inline docs/library/utime.rst]: s/API/ESP-IDF/ ?

**Review by tve — COMMENTED** [review]
Cool! I assume this is still worth merging? I don't remember your last experiences with the clock stuff, I believe you  were not convinced on the adjtime on the esp8266 but OK on the esp32, right?

**Comment by wieck** [comment]
There is no adjtime(3) in the ESP8266 IDF. As stated in the other (closed) pull request, that port needs mor work. This one I still think is worth merging.

File: docs/library/utime.rst (hunk 1 above)
Comment (tve) [reply docs/library/utime.rst]: mind fixing this? there are many APIs around whereas saying "the ESP-IDF's function" is unambiguous...

**Comment by dpgeorge** [comment]
This function can't be added to the `time` module because it's not in CPython.

If we were to add this (it does look useful!) then it could go in either `machine.RTC.adjtime()` (or maybe `esp32.adjtime()`).

**Pull request closed without merging**